core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -170,7 +172,6 @@ public void onGoAwaySent(int lastStreamId, long errorCode, ByteBuf debugData) {
     @Override
     public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData) {
         // Should not reuse a connection that received a GOAWAY frame.
-        HttpSession.get(channel()).deactivate();
         disconnectWhenFinished();
```

Comment:
Changed `disconnectWhenFinished()` to invoke `HttpSession.get(channel()).deactivate()` internally.